File: CS/WebSite/App_Code/ClientLayoutSerializer.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Web;
using System.Web.UI.WebControls;
using System.Xml;
using System.Xml.Serialization;
using DevExpress.Web.ASPxGridView;

public class ClientLayoutSerializer {
	private ASPxGridView grid;
	private const string SessionPostfix = "_layoutData";

	public ClientLayoutSerializer(ASPxGridView grid) {
		this.grid = grid;
	}

	public void RestoreClientLayout() {
		if (HttpContext.Current.Session[grid.ClientID + SessionPostfix] == null)
			return;
		string dataString = (String)HttpContext.Current.Session[grid.ClientID + SessionPostfix];
		DeserializeColumns(dataString);
	}
	public void SaveClientLayout() {
		string data = SerializeColumns();
		HttpContext.Current.Session[grid.ClientID + SessionPostfix] = data;
	}

	private string SerializeColumns() {
		StringBuilder sb = new StringBuilder();
		using (XmlWriter xw = XmlWriter.Create(sb)) {
			xw.WriteStartDocument();			

			GridViewInfo g = new GridViewInfo(grid);
			foreach (GridViewColumn column in grid.Columns) {
				if (column is GridViewDataColumn)			
					g.Columns.Add(new DataColumnInfo(column as GridViewDataColumn));
				else if (column is GridViewCommandColumn)			
					g.Columns.Add(new CommandColumnInfo(column as GridViewCommandColumn));
				else if (column is GridViewBandColumn)			
					g.Columns.Add(new BandColumnInfo(column as GridViewBandColumn));
			}			
			XmlSerializer xmlSerializer = new XmlSerializer(g.GetType());
			xmlSerializer.Serialize(xw, g);
		}
		return sb.ToString();
	}

	private void DeserializeColumns(string layoutData) {
		XmlDocument doc = new XmlDocument();
		doc.LoadXml(layoutData);

		GridViewInfo gridInfo = null;
		XmlSerializer xmlSerializer = new XmlSerializer(typeof(GridViewInfo));
		using (XmlReader xr = XmlReader.Create(new StringReader(layoutData))) {
			gridInfo = (GridViewInfo)xmlSerializer.Deserialize(xr);
		}
		gridInfo.RestoreGridViewLayout(grid);
	}
}

File: VB/WebSite/App_Code/ClientLayoutSerializer.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Web
Imports System.Web.UI.WebControls
Imports System.Xml
Imports System.Xml.Serialization
Imports DevExpress.Web.ASPxGridView

Public Class ClientLayoutSerializer
	Private grid As ASPxGridView
	Private Const SessionPostfix As String = "_layoutData"

	Public Sub New(ByVal grid As ASPxGridView)
		Me.grid = grid
	End Sub

	Public Sub RestoreClientLayout()
		If HttpContext.Current.Session(grid.ClientID & SessionPostfix) Is Nothing Then
			Return
		End If
		Dim dataString As String = CType(HttpContext.Current.Session(grid.ClientID & SessionPostfix), String)
		DeserializeColumns(dataString)
	End Sub
	Public Sub SaveClientLayout()
		Dim data As String = SerializeColumns()
		HttpContext.Current.Session(grid.ClientID & SessionPostfix) = data
	End Sub

	Private Function SerializeColumns() As String
		Dim sb As New StringBuilder()
		Using xw As XmlWriter = XmlWriter.Create(sb)
			xw.WriteStartDocument()

			Dim g As New GridViewInfo(grid)
			For Each column As GridViewColumn In grid.Columns
				If TypeOf column Is GridViewDataColumn Then
					g.Columns.Add(New DataColumnInfo(TryCast(column, GridViewDataColumn)))
				ElseIf TypeOf column Is GridViewCommandColumn Then
					g.Columns.Add(New CommandColumnInfo(TryCast(column, GridViewCommandColumn)))
				ElseIf TypeOf column Is GridViewBandColumn Then
					g.Columns.Add(New BandColumnInfo(TryCast(column, GridViewBandColumn)))
				End If
			Next column
			Dim xmlSerializer As New XmlSerializer(g.GetType())
			xmlSerializer.Serialize(xw, g)
		End Using
		Return sb.ToString()
	End Function

	Private Sub DeserializeColumns(ByVal layoutData As String)
		Dim doc As New XmlDocument()
		doc.LoadXml(layoutData)

		Dim gridInfo As GridViewInfo = Nothing
		Dim xmlSerializer As New XmlSerializer(GetType(GridViewInfo))
		Using xr As XmlReader = XmlReader.Create(New StringReader(layoutData))
			gridInfo = CType(xmlSerializer.Deserialize(xr), GridViewInfo)
		End Using
		gridInfo.RestoreGridViewLayout(grid)
	End Sub
End Class